Patches for v103

csagan5 opened the pull request:
Notable changes:
* excluding most of the Java code for signin, sync, commerce/shopping/price integrations (not yet excluding whole components)
* removed changes from files excluded in patches
* removed patch to enable HTTPS by default
* moved&re-created preference changes to `Modify-default-preferences.patch`

@uazo I have added an `Original License:` for all your patches with GPLv2, except those which I changed sensibly.

Currently compilation works but linking is broken; feel free to submit fixes if you have time, otherwise I will get to it next week.
There is still fixes needed on the C++ side and then there might be crashes.

You can also submit your patches improvements on top of these patches, since they will eventually be merged; I used in a few places your updated patches already but didn't have time to check for anything new.

# TODO
* [x] fix linking issues
* [x] fix any crash
* [ ] add any new patch from @uazo if relevant

uazo commented on build/patches/Reduce-HTTP-headers-in-DoH-requests-to-bare-minimum.patch:
```diff
@@ -11,11 +12,10 @@ Subject: Reduce HTTP headers in DoH requests to bare minimum
  // trusted process.
  LOAD_FLAG(CAN_USE_RESTRICTED_PREFETCH, 1 << 16)
-+
+ 
```
nothing relevant, but there are so many 'new line' here that can be eliminated

uazo commented on build/patches/Move-navigation-bar-to-bottom.patch:
```diff
@@ -538,32 +549,51 @@ diff --git a/chrome/android/java/src/org/chromium/chrome/browser/accessibility/s
+                     controlContainer.initWithToolbar(toolbarLayoutId);
++                    ImageView shadowImage = findViewById(R.id.toolbar_hairline);
++                    if (shadowImage != null) {
++                    if (CachedFeatureFlags.isEnabled(ChromeFeatureList.MOVE_TOP_TOOLBAR_TO_BOTTOM)) {
```
forgot the ident

uazo commented on build/patches/Enable-native-Android-autofill.patch:
```diff
@@ -11,45 +11,50 @@ along with browser-managed autofill.
-           : autofill::AutofillManager::AutofillManagerFactoryCallback());
+           ? base::BindRepeating(&autofill::AndroidDriverInitHook,
+                                 AwAutofillClient::FromWebContents(web_contents),
+-                                enable_download_manager)
```
I did it like this

```
diff --git a/android_webview/browser/aw_contents.cc b/android_webview/browser/aw_contents.cc
--- a/android_webview/browser/aw_contents.cc
+++ b/android_webview/browser/aw_contents.cc
@@ -336,6 +336,7 @@ void AwContents::InitAutofillIfNecessary(bool autocomplete_enabled) {
   // WebView browser tests may shall use BrowserAutofillManager if
   // `!autofill_provider`.
   ContentAutofillDriverFactory::DriverInitCallback driver_init_hook =
+      /*enable_browser_autofill_manager*/ false,
       autofill_provider
           ? base::BindRepeating(&autofill::AndroidDriverInitHook,
                                 AwAutofillClient::FromWebContents(web_contents),
```
however, I have not actually compiled the webview

csagan5 replied:
Yes, the `/*enable_browser_autofill_manager*/ false,` is out of place.

uazo commented on build/patches/Enable-IntentBlockExternalFormRedirectsNoGesture.patch:
```diff
@@ -2,6 +2,7 @@ From: csagan5 <32685696+csagan5@users.noreply.github.com>
 Date: Tue, 25 May 2021 19:33:41 +0200
 Subject: Enable IntentBlockExternalFormRedirectsNoGesture
 
+License: GPL-3.0-only - https://spdx.org/licenses/GPL-3.0-only.html
```
sure about this patch?
I modified it like this: https://github.com/uazo/bromite/blob/next-103/build/patches/Enable-IntentBlockExternalFormRedirectsNoGesture.patch

csagan5 replied:
The feature flag will not work?

uazo replied:
sorry, I didn't realize you did a revert.

I think `blockExternalFormRedirectsWithoutGesture` and `shouldDisableExternalIntentRequestsForUrl` do different things. I will check.

csagan5 replied:
Please let me know about this, I will merge this PR meanwhile.

uazo replied:
I double checked, my modification always blocks the opening regardless of the usergesture.
it is more correct as it is now.

csagan5 replied:
Ok, thanks - so I will not make any other change related to this.

uazo commented on build/patches/Disable-NTP-remote-suggestions-by-default.patch:
```diff
@@ -2,42 +2,12 @@ From: csagan5 <32685696+csagan5@users.noreply.github.com>
--        if (mFeedEnabled == !mTabToStreamMap.isEmpty()) {
--            return;
--        }
-+        // Bromite needs the next lines removed because mFeedEnabled
```
it seems to me that this whole part is missing

csagan5 replied:
The feed Java files are not built, they are excluded with modifications to `BUILD.gn` files.

uazo replied:
> The feed Java files are not built, they are excluded with modifications to BUILD.gn files

yes, you are right, I modified them before removing it from the gn's

uazo commented on build/patches/Disable-all-predictors-code.patch:
```diff
@@ -2,20 +2,22 @@ From: uazo <uazo@users.noreply.github.com>
  .../segmentation_platform/public/features.cc  |  2 +-
  third_party/blink/common/features.cc          |  2 +-
- 12 files changed, 25 insertions(+), 20 deletions(-)
+ 12 files changed, 27 insertions(+), 22 deletions(-)
```
we could add this:
```
diff --git a/components/optimization_guide/core/prediction_model_download_manager.cc b/components/optimization_guide/core/prediction_model_download_manager.cc
--- a/components/optimization_guide/core/prediction_model_download_manager.cc
+++ b/components/optimization_guide/core/prediction_model_download_manager.cc
@@ -96,7 +96,7 @@ PredictionModelDownloadManager::PredictionModelDownloadManager(
     const base::FilePath& models_dir_path,
     scoped_refptr<base::SequencedTaskRunner> background_task_runner)
     : download_service_(download_service),
-      is_available_for_downloads_(true),
+      is_available_for_downloads_(false),
       api_key_(features::GetOptimizationGuideServiceAPIKey()),
       models_dir_path_(models_dir_path),
       background_task_runner_(background_task_runner) {}
```

uazo commented on build/patches/Disable-all-predictors-code.patch:
```diff
@@ -175,22 +179,22 @@ diff --git a/components/optimization_guide/core/prediction_model_download_manage
-     const base::FilePath& file_path) {
+ PredictionModelDownloadManager::VerifyDownload(const base::FilePath& file_path,
+                                                bool delete_file_on_error) {
 +  if ((true)) return absl::nullopt;
```
i think file deletion is missing

```
@@ -247,6 +248,12 @@ void PredictionModelDownloadManager::OnDownloadFailed(
 absl::optional<std::pair<base::FilePath, base::FilePath>>
 PredictionModelDownloadManager::VerifyDownload(const base::FilePath& file_path,
                                                bool delete_file_on_error) {
+  if ((true)) {
+    base::ThreadPool::PostTask(
+        FROM_HERE, {base::TaskPriority::BEST_EFFORT, base::MayBlock()},
+        base::GetDeleteFileCallback(file_path));
+    return absl::nullopt;
+  }
   if (!switches::ShouldSkipModelDownloadVerificationForTesting()) {
     // Verify that the |file_path| contains a valid CRX file.
     std::string public_key;
```

csagan5 replied:
Why should it be deleted?

csagan5 replied:
We shouldn't even download it if we are going to delete it. I think it is not downloaded.

uazo replied:
> I think it is not downloaded.

sure, but it doesn't get there then and that `if ((true)) return absl::nullopt;` isn't necessary.
if it arrives there for some reason, the file is present and must be deleted

uazo commented on build/patches/Block-gateway-attacks-via-websockets.patch:
```diff
@@ -45,14 +47,14 @@ diff --git a/third_party/blink/renderer/core/execution_context/execution_context
+@@ -378,6 +378,7 @@ class CORE_EXPORT ExecutionContext : public Supplementable<ExecutionContext>,
+   void SetAddressSpace(network::mojom::blink::IPAddressSpace ip_address_space);
  
+   HeapObserverSet<ContextLifecycleObserver>& ContextLifecycleObserverSet();
```
here you should miss the include
```
diff --git a/third_party/blink/renderer/core/execution_context/execution_context.h b/third_party/blink/renderer/core/execution_context/execution_context.h
--- a/third_party/blink/renderer/core/execution_context/execution_context.h
+++ b/third_party/blink/renderer/core/execution_context/execution_context.h
@@ -33,6 +33,7 @@
 #include "base/notreached.h"
 #include "services/metrics/public/cpp/ukm_source_id.h"
 #include "services/network/public/mojom/referrer_policy.mojom-blink-forward.h"
+#include "services/network/public/mojom/ip_address_space.mojom-blink-forward.h"
 #include "third_party/abseil-cpp/absl/types/optional.h"
 #include "third_party/blink/public/common/tokens/tokens.h"
 #include "third_party/blink/public/common/user_agent/user_agent_metadata.h"
@@ -373,6 +374,8 @@ class CORE_EXPORT ExecutionContext : public Supplementable<ExecutionContext>,
       const String& message = g_empty_string,
       const String& source_file = g_empty_string) const {}
 
+  String addressSpaceForBindings() const;
+  network::mojom::IPAddressSpace AddressSpace() const;

```

csagan5 replied:
Your diff adds the include, but it is already there. I don't understand?

uazo replied:
> Your diff adds the include, but it is already there. I don't understand?

you have `Revert-removal-of-execution-context-address-space.patch` that I didn't have

uazo commented:
checked the patches you kept between releases.
I also saw that you are following up on https://github.com/bromite/bromite/issues/2141

> add any new patch from @uazo if relevant

please, take [Add-option-to-force-tablet-UI](https://github.com/uazo/bromite/blob/next-103/build/patches/Add-option-to-force-tablet-UI.patch), otherwise i can't compile in debug.
instead if you want [Disable-conversion-measurement-api.patch](https://github.com/uazo/bromite/blob/next-103/build/patches/Disable-conversion-measurement-api.patch) removes response-header-based registration

uazo commented:
> fix linking issues

do you still have it? do you want me to check?

uazo commented:
> See if you can fix the linking issues?

I've lauched the [build](https://github.com/uazo/bromite-buildtools/actions/runs/2574206588), I will let you know

csagan5 commented:
> > excluding most of the Java code for signin, sync, commerce/shopping/price integrations (not yet excluding whole components)
> 
> the problem lies here, in particular due to the heavy exclusions from `chrome/browser/BUILD.gn` are you sure of the road you are taking? the elimination of every class factory is an important part, and carries the risk of an unmanageable patch, or bugs related to null services (if we do not re-enter the various clients). i can try to get rid of all link problems by re-entering the factories, but i think it will crash in many places.

It is not the first time I try this approach, but every time I give up for similar reasons; this time I saw that basically every single network request gets associated to identity.

> > It is already there since a few major releases.
> 
> missing however [this](https://gist.github.com/uazo/c8e707d31fc96401cc238b62ceeb9182)

The option to force tablet helps with preventing crashes?



> > i can try to get rid of all link problems by re-entering the factories, but i think it will crash in many places.
> 
> with [this](https://gist.github.com/uazo/7d44fbd828419a5e133fcb6a96d2ae2a) you can link and start without crashing, but there may be other crashes.

Thanks, I will compare with my version; I will have to go back to an earlier version and keep only the changes on Java-side.

uazo commented:
> The option to force tablet helps with preventing crashes?

I don't think, this is the build error
```
============================= Dependency Checks Failed =============================
Target: //components/browser_ui/accessibility/android:java
Direct classpath is incomplete. To fix, add deps on:
 * //chrome/browser/preferences:java
     * org.chromium.chrome.browser.preferences.ChromePreferenceKeys (needed by org.chromium.components.browser_ui.accessibility.AccessibilitySettings)
     * org.chromium.chrome.browser.preferences.SharedPreferencesManager (needed by org.chromium.components.browser_ui.accessibility.AccessibilitySettings)
Traceback (most recent call last):
  File "../../build/android/gyp/bytecode_processor.py", line 82, in <module>
    sys.exit(main(sys.argv))
  File "../../build/android/gyp/bytecode_processor.py", line 72, in main
    build_utils.CheckOutput(cmd,
  File "/home/lg/working_dir/chromium/src/build/android/gyp/util/build_utils.py", line 294, in CheckOutput
    raise CalledProcessError(cwd, args, MSG.format(stream_string))
util.build_utils.CalledProcessError: Command failed: ( cd /home/lg/working_dir/chromium/src/out/bromite; bin/helper/bytecode_processor //components/browser_ui/accessibility/android:java 
```
```
You can often set treat_warnings_as_errors=false to not treat output as failure (useful when developing locally).
```
source https://github.com/uazo/bromite-buildtools/runs/7087412576?check_suite_focus=true#step:4:39365
I have `treat_warnings_as_errors=true` because in debug

>  I saw that basically every single network request gets associated to identity.

Where?

csagan5 commented:
> I don't think, this is the build error

Ok, but that patch does not mention the issue. Can you please submit the patch in a PR with proper description? Also: any other patch you would like to submit? I am getting close to a working build, had to redo the signin/sync removal patch (I think I did it 3 times so far for this release).

> I have `treat_warnings_as_errors=true` because in debug

Would this be useful in general to avoid crashes and bugs? Maybe we could enable it for the official Bromite build too.

> > I saw that basically every single network request gets associated to identity.
> 
> Where?

On the C++ side; last time I checked (couple years ago) I don't remember it being so pervasive. It's nothing new to worry about but I think - like for network traffic annotations - they have on server side also annotations for the users.

uazo commented:
> Can you please submit the patch in a PR with proper description?

sure, but I thought of integrating it into the same (as a fixup to understand).

> Would this be useful in general to avoid crashes and bugs? 

I don't know, the linking phase of the apk is something I don't know.
in my opinion no, I imagine that in the end each module ends up in the same file (the apk in fact, java side), but perhaps that choice was introduced because with another target we would have a cyclic reference.

> Also: any other patch you would like to submit?

I already have some fix ready, but I haven't made them available yet, I was waiting for a definitive v103 build, do you prefer to have them available?
others are new things to discuss, but I didn't want to rush you

csagan5 commented:
> I don't know, the linking phase of the apk is something I don't know.
> in my opinion no, I imagine that in the end each module ends up in the same file (the apk in fact, java side), but perhaps that choice was introduced because with another target we would have a cyclic reference.

I think there might be runtime errors without this check.

> I already have some fix ready, but I haven't made them available yet, I was waiting for a definitive v103 build, do you prefer to have them available?

Most of the patches are already definitive, only a couple big ones need some adjustments; yes, the patches can be submitted when you want, there is no need to wait.

> others are new things to discuss, but I didn't want to rush you

No rush, we can discuss things, it does not affect the release pace.

uazo commented:
>  yes, the patches can be submitted when you want, there is no need to wait.

I'm afraid of losing my changes if we proceed that way, simply because I haven't aligned my branch with yours.
I'm starting to make fixups available to you:

- https://gist.github.com/uazo/d07da555d73e5c4f6fdbb3c5b23c1526
those fix https://github.com/bromite/bromite/issues/2133#issuecomment-1172277504

- https://gist.github.com/uazo/3e46ffe8626b09864102fe4091c52559 
fix #2096 

the new patches instead I can prepare them, so let's talk about them.

csagan5 commented:
> > yes, the patches can be submitted when you want, there is no need to wait.
> 
> I'm afraid of losing my changes if we proceed that way, simply because I haven't aligned my branch with yours.

I was referring to new patches not updates to existing patches. By the way, the android autofill patch has been giving me lots of troubles, how did you build it on your branch?

uazo commented:
> By the way, the android autofill patch has been giving me lots of troubles, how did you build it on your branch?

Haven't you seen this one? or doesn't it work for you?
https://github.com/uazo/bromite/blob/next-103/build/patches/Enable-native-Android-autofill.patch
because it works for me:
![image](https://user-images.githubusercontent.com/29201891/177939773-1f5b0110-6516-4376-bd9f-59c4c1ce5845.png)

csagan5 commented:
> Haven't you seen this one? or doesn't it work for you?
> https://github.com/uazo/bromite/blob/next-103/build/patches/Enable-native-Android-autofill.patch
> because it works for me:

This does not compile for the SystemWebView:
```
diff --git a/android_webview/browser/aw_contents.cc b/android_webview/browser/aw_contents.cc
--- a/android_webview/browser/aw_contents.cc
+++ b/android_webview/browser/aw_contents.cc
@@ -336,6 +336,7 @@ void AwContents::InitAutofillIfNecessary(bool autocomplete_enabled) {
   // WebView browser tests may shall use BrowserAutofillManager if
   // `!autofill_provider`.
   ContentAutofillDriverFactory::DriverInitCallback driver_init_hook =
+      /*enable_browser_autofill_manager*/ false,
       autofill_provider
           ? base::BindRepeating(&autofill::AndroidDriverInitHook,
                                 AwAutofillClient::FromWebContents(web_contents),
```

uazo commented:
> This does not compile for the SystemWebView

sorry I did not understand, I understood it worked!
I launched the build, I fix it and I'll let you know.

uazo commented:
very sorry,  was the other call to be changed ...
so it build:
```
  // WebView browser tests may shall use BrowserAutofillManager if
  // `!autofill_provider`.
  ContentAutofillDriverFactory::DriverInitCallback driver_init_hook =
      autofill_provider
          ? base::BindRepeating(&autofill::AndroidDriverInitHook,
                                AwAutofillClient::FromWebContents(web_contents),
                                enable_download_manager)                           <---- not here
          : base::BindRepeating(&autofill::BrowserDriverInitHook,
                                AwAutofillClient::FromWebContents(web_contents),
                                base::android::GetDefaultLocaleString(),
                                /*enable_secondary_autofill_manager*/ false);      <---- here
```

csagan5 commented:
No worries; I thought I had already tried that, but perhaps I never removed it from the first one.

uazo commented:
it took me a while to understand how `base::BindRepeating` work.. I was hoping I caught it on the first try, but unfortunately... 
I have to add the webview build in my tests
again, sorry if I wasted your time

uazo commented:
ops.. now I have:
```
[8832/25024] ACTION //v8:generate_bytecode_builtins_list(//build/toolchain/android:android_clang_x86)
FAILED: android_clang_x86/gen/v8/builtins-generated/bytecodes-builtins-list.h
python3 ../../v8/tools/run.py ./clang_x86/bytecode_builtins_list_generator android_clang_x86/gen/v8/builtins-generated/bytecodes-builtins-list.h

Return code is -31 
```
have you already solved it? do you want me to check?

csagan5 commented:
> have you already solved it? do you want me to check?

I am solving crash issues now; in the past I was able to use `third_party/android_platform/development/scripts/stack.py` and feed it the crash dumps from `adb logcat` to see where the crash happened. How do you go from an `adb` crash dump to a code location nowadays? And also: have you been able to use tombstones? They have other similar issues.

uazo commented:
maybe I misunderstood but

>  in the past I was able to use

you mean this?
```
adb root
adb shell ls /data/tombstones/ -l
TOMBSTONE=xxx && adb pull /data/tombstones/$TOMBSTONE
third_party/android_platform/development/scripts/stack --output-directory out/bromite/ $TOMBSTONE >$TOMBSTONE.txt
```

> How do you go from an adb crash dump to a code location nowadays? 

this is one of the results (one of many I do :)
```
Build fingerprint: 'Android/sdk_phone_x86_64/generic_x86_64:9/PSR1.180720.012/4923214:userdebug/test-keys'
Revision: '0'
pid: 9784, tid: 9784, name: ite.bromite.dev  >>> org.bromite.bromite.dev <<<
signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0xb8

Stack Trace:
  RELADDR   FUNCTION                                                                          FILE:LINE
  v------>  std::__1::vector<base::internal::UncheckedObserverAdapter, std::__1::allocator<base::internal::UncheckedObserverAdapter>>::begin() const  ../../buildtools/third_party/libc++/trunk/include/vector:1526:30
  v------>  decltype(std::forward<std::__1::vector<base::internal::UncheckedObserverAdapter, std::__1::allocator<base::internal::UncheckedObserverAdapter>> const&>(fp).begin()) base::internal::begin<std::__1::vector<base::internal::UncheckedObserverAdapter, std::__1::allocator<base::internal::UncheckedObserverAdapter>> const&>(std::__1::vector<base::internal::UncheckedObserverAdapter, std::__1::allocator<base::internal::UncheckedObserverAdapter>> const&, base::internal::priority_tag<1ul>)  ../../base/ranges/ranges.h:44:37
  v------>  decltype(internal::begin(std::forward<std::__1::vector<base::internal::UncheckedObserverAdapter, std::__1::allocator<base::internal::UncheckedObserverAdapter>> const&>(fp), (base::internal::priority_tag<2ul>)())) base::ranges::begin<std::__1::vector<base::internal::UncheckedObserverAdapter, std::__1::allocator<base::internal::UncheckedObserverAdapter>> const&>(std::__1::vector<base::internal::UncheckedObserverAdapter, std::__1::allocator<base::internal::UncheckedObserverAdapter>> const&)  ../../base/ranges/ranges.h:105:10
  0000000005a6126b  auto base::ranges::find_if<std::__1::vector<base::internal::UncheckedObserverAdapter, std::__1::allocator<base::internal::UncheckedObserverAdapter>> const&, base::ObserverList<signin::IdentityManager::Observer, true, true, base::internal::UncheckedObserverAdapter>::HasObserver(signin::IdentityManager::Observer const*) const::'lambda'(auto const&), base::identity, std::__1::random_access_iterator_tag>(auto&&, base::ObserverList<signin::IdentityManager::Observer, true, true, base::internal::UncheckedObserverAdapter>::HasObserver(signin::IdentityManager::Observer const*) const::'lambda'(auto const&), base::identity)  ../../base/ranges/algorithm.h:483:26
  0000000005a6125b  base::ObserverList<signin::IdentityManager::Observer, true, true, base::internal::UncheckedObserverAdapter>::HasObserver(signin::IdentityManager::Observer const*) const  ../../base/observer_list.h:307:12
  0000000005a6058e  base::ObserverList<signin::IdentityManager::Observer, true, true, base::internal::UncheckedObserverAdapter>::AddObserver(signin::IdentityManager::Observer*)  ../../base/observer_list.h:274:9
  0000000005a62a1c  base::ScopedObservation<signin::IdentityManager, signin::IdentityManager::Observer, &signin::IdentityManager::AddObserver(signin::IdentityManager::Observer*), &signin::IdentityManager::RemoveObserver(signin::IdentityManager::Observer*)>::Observe(signin::IdentityManager*)  ../../base/scoped_observation.h:63:5
  000000000aa964ac  syncer::SyncSessionDurationsMetricsRecorder::SyncSessionDurationsMetricsRecorder(syncer::SyncService*, signin::IdentityManager*)  ../../components/sync/driver/sync_session_durations_metrics_recorder.cc:45:33
  0000000008d5d1e8  std::__1::__unique_if<syncer::SyncSessionDurationsMetricsRecorder>::__unique_single std::__1::make_unique<syncer::SyncSessionDurationsMetricsRecorder, syncer::SyncService*&, signin::IdentityManager*&>(syncer::SyncService*&, signin::IdentityManager*&)  ../../buildtools/third_party/libc++/trunk/include/__memory/unique_ptr.h:725:32
  0000000008d5d184  AndroidSessionDurationsService::InitializeForRegularProfile(syncer::SyncService*, signin::IdentityManager*)  ../../chrome/browser/android/metrics/android_session_durations_service.cc:92:7
  0000000008d5d802  AndroidSessionDurationsServiceFactory::BuildServiceInstanceFor(content::BrowserContext*) const  ../../chrome/browser/android/metrics/android_session_durations_service_factory.cc:92:12
  000000000a2b68ca  KeyedServiceFactory::GetServiceForContext(void*, bool)                            ../../components/keyed_service/core/keyed_service_factory.cc:80:15
  0000000008d5d664  (anonymous namespace)::GetForAllActiveProfiles()                                  ../../chrome/browser/android/metrics/android_session_durations_service_factory.cc:25:7
```
the position is present.
however my build is "debug", maybe in "release" it behaves differently

> And also: have you been able to use tombstones? 

I use them often, indeed, every now and then I insert a `DCHECK (false)` on purpose to understand the stack trace

csagan5 commented:
There are two approaches:
1. copy/pasting the crash dump from `adb logcat` into a file and passing it to `stack.py`
2. passing tombstones to `stack.py`

They both used to work; I was referring to (1) specifically (see also: https://source.android.com/devices/tech/debug#debuggerd). (2) seems to work but the Java part seems not unaliased:
```
Stack Trace:
  RELADDR   FUNCTION                                                                          FILE:LINE
  03b16513  offset 0x17d5000) (BuildId: 376fbbcaef08cf4d5b7b8143001efcb6a8a105bb              /data/app/~~uPRM59E4hWa0OD66xUBVMQ==/org.bromite.bromite-JO6vHGgDoqRWEdQQwGup2w==/base.apk/libchrome.so
  03bac642  offset 0x17d5000) (BuildId: 376fbbcaef08cf4d5b7b8143001efcb6a8a105bb              /data/app/~~uPRM59E4hWa0OD66xUBVMQ==/org.bromite.bromite-JO6vHGgDoqRWEdQQwGup2w==/base.apk/libchrome.so
  020aa4b6  offset 0x17d5000) (BuildId: 376fbbcaef08cf4d5b7b8143001efcb6a8a105bb              /data/app/~~uPRM59E4hWa0OD66xUBVMQ==/org.bromite.bromite-JO6vHGgDoqRWEdQQwGup2w==/base.apk/libchrome.so
  02650cd8  offset 0x17d5000) (BuildId: 376fbbcaef08cf4d5b7b8143001efcb6a8a105bb              /data/app/~~uPRM59E4hWa0OD66xUBVMQ==/org.bromite.bromite-JO6vHGgDoqRWEdQQwGup2w==/base.apk/libchrome.so
  0264f049  offset 0x17d5000) (BuildId: 376fbbcaef08cf4d5b7b8143001efcb6a8a105bb              /data/app/~~uPRM59E4hWa0OD66xUBVMQ==/org.bromite.bromite-JO6vHGgDoqRWEdQQwGup2w==/base.apk/libchrome.so
  0263cf3f  offset 0x17d5000) (BuildId: 376fbbcaef08cf4d5b7b8143001efcb6a8a105bb              /data/app/~~uPRM59E4hWa0OD66xUBVMQ==/org.bromite.bromite-JO6vHGgDoqRWEdQQwGup2w==/base.apk/libchrome.so
  03a11833  offset 0x17d5000) (BuildId: 376fbbcaef08cf4d5b7b8143001efcb6a8a105bb              /data/app/~~uPRM59E4hWa0OD66xUBVMQ==/org.bromite.bromite-JO6vHGgDoqRWEdQQwGup2w==/base.apk/libchrome.so
  052f93e8  offset 0x17d5000) (BuildId: 376fbbcaef08cf4d5b7b8143001efcb6a8a105bb              /data/app/~~uPRM59E4hWa0OD66xUBVMQ==/org.bromite.bromite-JO6vHGgDoqRWEdQQwGup2w==/base.apk/libchrome.so
  0495e63e  offset 0x17d5000) (BuildId: 376fbbcaef08cf4d5b7b8143001efcb6a8a105bb              /data/app/~~uPRM59E4hWa0OD66xUBVMQ==/org.bromite.bromite-JO6vHGgDoqRWEdQQwGup2w==/base.apk/libchrome.so
  03a115a2  offset 0x17d5000) (BuildId: 376fbbcaef08cf4d5b7b8143001efcb6a8a105bb              /data/app/~~uPRM59E4hWa0OD66xUBVMQ==/org.bromite.bromite-JO6vHGgDoqRWEdQQwGup2w==/base.apk/libchrome.so
  03a0dd2d  offset 0x17d5000) (BuildId: 376fbbcaef08cf4d5b7b8143001efcb6a8a105bb              /data/app/~~uPRM59E4hWa0OD66xUBVMQ==/org.bromite.bromite-JO6vHGgDoqRWEdQQwGup2w==/base.apk/libchrome.so
  052f93e8  offset 0x17d5000) (BuildId: 376fbbcaef08cf4d5b7b8143001efcb6a8a105bb              /data/app/~~uPRM59E4hWa0OD66xUBVMQ==/org.bromite.bromite-JO6vHGgDoqRWEdQQwGup2w==/base.apk/libchrome.so
  0495e63e  offset 0x17d5000) (BuildId: 376fbbcaef08cf4d5b7b8143001efcb6a8a105bb              /data/app/~~uPRM59E4hWa0OD66xUBVMQ==/org.bromite.bromite-JO6vHGgDoqRWEdQQwGup2w==/base.apk/libchrome.so
  052f9416  offset 0x17d5000) (BuildId: 376fbbcaef08cf4d5b7b8143001efcb6a8a105bb              /data/app/~~uPRM59E4hWa0OD66xUBVMQ==/org.bromite.bromite-JO6vHGgDoqRWEdQQwGup2w==/base.apk/libchrome.so
  0495e025  offset 0x17d5000) (BuildId: 376fbbcaef08cf4d5b7b8143001efcb6a8a105bb              /data/app/~~uPRM59E4hWa0OD66xUBVMQ==/org.bromite.bromite-JO6vHGgDoqRWEdQQwGup2w==/base.apk/libchrome.so
  052f902b  offset 0x17d5000) (BuildId: 376fbbcaef08cf4d5b7b8143001efcb6a8a105bb              /data/app/~~uPRM59E4hWa0OD66xUBVMQ==/org.bromite.bromite-JO6vHGgDoqRWEdQQwGup2w==/base.apk/libchrome.so
  052f8fc2  offset 0x17d5000) (BuildId: 376fbbcaef08cf4d5b7b8143001efcb6a8a105bb              /data/app/~~uPRM59E4hWa0OD66xUBVMQ==/org.bromite.bromite-JO6vHGgDoqRWEdQQwGup2w==/base.apk/libchrome.so
  039ee68b  offset 0x17d5000) (BuildId: 376fbbcaef08cf4d5b7b8143001efcb6a8a105bb              /data/app/~~uPRM59E4hWa0OD66xUBVMQ==/org.bromite.bromite-JO6vHGgDoqRWEdQQwGup2w==/base.apk/libchrome.so
  039f2c97  offset 0x17d5000) (BuildId: 376fbbcaef08cf4d5b7b8143001efcb6a8a105bb              /data/app/~~uPRM59E4hWa0OD66xUBVMQ==/org.bromite.bromite-JO6vHGgDoqRWEdQQwGup2w==/base.apk/libchrome.so
  039f11f6  offset 0x17d5000) (BuildId: 376fbbcaef08cf4d5b7b8143001efcb6a8a105bb              /data/app/~~uPRM59E4hWa0OD66xUBVMQ==/org.bromite.bromite-JO6vHGgDoqRWEdQQwGup2w==/base.apk/libchrome.so
  039f18b6  offset 0x17d5000) (BuildId: 376fbbcaef08cf4d5b7b8143001efcb6a8a105bb              /data/app/~~uPRM59E4hWa0OD66xUBVMQ==/org.bromite.bromite-JO6vHGgDoqRWEdQQwGup2w==/base.apk/libchrome.so
  038f4ef9  offset 0x17d5000) (BuildId: 376fbbcaef08cf4d5b7b8143001efcb6a8a105bb              /data/app/~~uPRM59E4hWa0OD66xUBVMQ==/org.bromite.bromite-JO6vHGgDoqRWEdQQwGup2w==/base.apk/libchrome.so
  027953c5  offset 0x17d5000) (BuildId: 376fbbcaef08cf4d5b7b8143001efcb6a8a105bb              /data/app/~~uPRM59E4hWa0OD66xUBVMQ==/org.bromite.bromite-JO6vHGgDoqRWEdQQwGup2w==/base.apk/libchrome.so
  01d85d2b  offset 0x17d5000) (BuildId: 376fbbcaef08cf4d5b7b8143001efcb6a8a105bb              /data/app/~~uPRM59E4hWa0OD66xUBVMQ==/org.bromite.bromite-JO6vHGgDoqRWEdQQwGup2w==/base.apk/libchrome.so
  02407975  offset 0x17d5000) (BuildId: 376fbbcaef08cf4d5b7b8143001efcb6a8a105bb              /data/app/~~uPRM59E4hWa0OD66xUBVMQ==/org.bromite.bromite-JO6vHGgDoqRWEdQQwGup2w==/base.apk/libchrome.so
  02b147bf  offset 0x17d5000) (BuildId: 376fbbcaef08cf4d5b7b8143001efcb6a8a105bb              /data/app/~~uPRM59E4hWa0OD66xUBVMQ==/org.bromite.bromite-JO6vHGgDoqRWEdQQwGup2w==/base.apk/libchrome.so
  01d85d2b  offset 0x17d5000) (BuildId: 376fbbcaef08cf4d5b7b8143001efcb6a8a105bb              /data/app/~~uPRM59E4hWa0OD66xUBVMQ==/org.bromite.bromite-JO6vHGgDoqRWEdQQwGup2w==/base.apk/libchrome.so
  03b546bf  offset 0x17d5000) (BuildId: 376fbbcaef08cf4d5b7b8143001efcb6a8a105bb              /data/app/~~uPRM59E4hWa0OD66xUBVMQ==/org.bromite.bromite-JO6vHGgDoqRWEdQQwGup2w==/base.apk/libchrome.so
  03b6a04d  offset 0x17d5000) (BuildId: 376fbbcaef08cf4d5b7b8143001efcb6a8a105bb              /data/app/~~uPRM59E4hWa0OD66xUBVMQ==/org.bromite.bromite-JO6vHGgDoqRWEdQQwGup2w==/base.apk/libchrome.so
  03b6a773  offset 0x17d5000) (BuildId: 376fbbcaef08cf4d5b7b8143001efcb6a8a105bb              /data/app/~~uPRM59E4hWa0OD66xUBVMQ==/org.bromite.bromite-JO6vHGgDoqRWEdQQwGup2w==/base.apk/libchrome.so
  03ba1166  offset 0x17d5000) (BuildId: 376fbbcaef08cf4d5b7b8143001efcb6a8a105bb              /data/app/~~uPRM59E4hWa0OD66xUBVMQ==/org.bromite.bromite-JO6vHGgDoqRWEdQQwGup2w==/base.apk/libchrome.so
  03ba110e  offset 0x17d5000) (BuildId: 376fbbcaef08cf4d5b7b8143001efcb6a8a105bb              /data/app/~~uPRM59E4hWa0OD66xUBVMQ==/org.bromite.bromite-JO6vHGgDoqRWEdQQwGup2w==/base.apk/libchrome.so
  03ba0df5  offset 0x17d5000) (BuildId: 376fbbcaef08cf4d5b7b8143001efcb6a8a105bb              /data/app/~~uPRM59E4hWa0OD66xUBVMQ==/org.bromite.bromite-JO6vHGgDoqRWEdQQwGup2w==/base.apk/libchrome.so
  00019337  android::SimpleLooperCallback::handleEvent(int, int, void*)+39) (BuildId: ab4be013cda31e8c45d48aa23a89d0f8  /system/lib/libutils.so
  0001a367  android::Looper::pollInner(int)+1127) (BuildId: ab4be013cda31e8c45d48aa23a89d0f8  /system/lib/libutils.so
  00019e96  android::Looper::pollOnce(int, int*, int*, void**)+118) (BuildId: ab4be013cda31e8c45d48aa23a89d0f8  /system/lib/libutils.so
  0010ef8b  android::android_os_MessageQueue_nativePollOnce(_JNIEnv*, _jobject*, long long, int)+59) (BuildId: 588f2cd5873ff4273bb25b25edb82606  /system/lib/libandroid_runtime.so
  0021be48  art_jni_trampoline+136) (BuildId: 9a9778e61b43d349325d0bb85244bd9bc95ff387        /system/framework/x86/boot-framework.oat
  00639e80  android.os.MessageQueue.next+224) (BuildId: 9a9778e61b43d349325d0bb85244bd9bc95ff387  /system/framework/x86/boot-framework.oat
  00636550  android.os.Looper.loop+592) (BuildId: 9a9778e61b43d349325d0bb85244bd9bc95ff387    /system/framework/x86/boot-framework.oat
  0040581a  android.app.ActivityThread.main+746) (BuildId: 9a9778e61b43d349325d0bb85244bd9bc95ff387  /system/framework/x86/boot-framework.oat
  0013baf2  art_quick_invoke_static_stub+418) (BuildId: 8191579dfafff37a5cbca70f9a73020f      /apex/com.android.art/lib/libart.so
  001d0392  art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+258) (BuildId: 8191579dfafff37a5cbca70f9a73020f  /apex/com.android.art/lib/libart.so
  00630008  art::InvokeMethod(art::ScopedObjectAccessAlreadyRunnable const&, _jobject*, _jobject*, _jobject*, unsigned int)+1464) (BuildId: 8191579dfafff37a5cbca70f9a73020f  /apex/com.android.art/lib/libart.so
  005886a0  art::Method_invoke(_JNIEnv*, _jobject*, _jobject*, _jobjectArray*)+80) (BuildId: 8191579dfafff37a5cbca70f9a73020f  /apex/com.android.art/lib/libart.so
  0008a778  art_jni_trampoline+168) (BuildId: b386118502376442aa402e8d28c29e95a6eb8614        /apex/com.android.art/javalib/x86/boot.oat
  0088ac49  com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run+137) (BuildId: 9a9778e61b43d349325d0bb85244bd9bc95ff387  /system/framework/x86/boot-framework.oat
  00893656  com.android.internal.os.ZygoteInit.main+2102) (BuildId: 9a9778e61b43d349325d0bb85244bd9bc95ff387  /system/framework/x86/boot-framework.oat
  0013baf2  art_quick_invoke_static_stub+418) (BuildId: 8191579dfafff37a5cbca70f9a73020f      /apex/com.android.art/lib/libart.so
  001d0392  art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+258) (BuildId: 8191579dfafff37a5cbca70f9a73020f  /apex/com.android.art/lib/libart.so
  0062e653  art::JValue art::InvokeWithVarArgs<art::ArtMethod*>(art::ScopedObjectAccessAlreadyRunnable const&, _jobject*, art::ArtMethod*, char*)+579) (BuildId: 8191579dfafff37a5cbca70f9a73020f  /apex/com.android.art/lib/libart.so
  0062eb25  art::JValue art::InvokeWithVarArgs<_jmethodID*>(art::ScopedObjectAccessAlreadyRunnable const&, _jobject*, _jmethodID*, char*)+85) (BuildId: 8191579dfafff37a5cbca70f9a73020f  /apex/com.android.art/lib/libart.so
  004ce64f  art::JNI<true>::CallStaticVoidMethodV(_JNIEnv*, _jclass*, _jmethodID*, char*)+735) (BuildId: 8191579dfafff37a5cbca70f9a73020f  /apex/com.android.art/lib/libart.so
  0008f90e  _JNIEnv::CallStaticVoidMethod(_jclass*, _jmethodID*, ...)+62) (BuildId: 588f2cd5873ff4273bb25b25edb82606  /system/lib/libandroid_runtime.so
  00098c8e  android::AndroidRuntime::start(char const*, android::Vector<android::String8> const&, bool)+910) (BuildId: 588f2cd5873ff4273bb25b25edb82606  /system/lib/libandroid_runtime.so
  00003804  main+1588) (BuildId: c5eedbfb6130af84c3db8e121fb1202e                             /system/bin/app_process32
  000522e3  __libc_init+115) (BuildId: 6e3a0180fa6637b68c0d181c343e6806                       /apex/com.android.runtime/lib/bionic/libc.so
```

> adb root
> adb shell ls /data/tombstones/ -l
> TOMBSTONE=xxx && adb pull /data/tombstones/$TOMBSTONE
> third_party/android_platform/development/scripts/stack --output-directory out/bromite/ $TOMBSTONE >$TOMBSTONE.txt

As mentioned before AVDs are not rooted, this cannot be used.

> however my build is "debug", maybe in "release" it behaves differently

I used to be able to do this also with the release.

Right now I see a crash on startup so I am debugging that; you are using same patches as in this branch and do not have it?

uazo commented:
> As mentioned before AVDs are not rooted, this cannot be used.

I don't know if they are, but `adb root` works, at least the ones you download with Android Studio.
how come it doesn't work for you?

> seems to work but the Java part seems not unaliased:

Some time ago, I had read about it on the chromium-dev mailing list, but as it didn't happen to me I didn't investigate. try looking there.

> Right now I see a crash on startup so I am debugging that; you are using same patches as in this branch and do not have it?

no, unfortunately I'm still on your second version. do you have any tools that compare applied patches and are able to produce some sort of diff to apply?
at most I throw everything and start again (for the third time :), I'll do some other pulls (sorry if I clog the repo) so I download the changes/new patches I'm doing.
but if you want to pass the crash maybe I can help you too

uazo commented:
```
001d0392  art::ArtMethod::Invoke(art::Thread*, unsigned int*, unsigned int, art::JValue*, char const*)+258) 
(BuildId: 8191579dfafff37a5cbca70f9a73020f  /apex/com.android.art/lib/libart.so
```
that type of crash has happened to me before, and, by heart, it was due to the linker, for the lack of some reference in the gn dependencies. maybe you have removed too much.
I launch the build, but it takes at least 3 hours.

csagan5 commented:
> I don't know if they are, but `adb root` works, at least the ones you download with Android Studio.
> how come it doesn't work for you?

It says `adbd cannot run as root in production builds`; the versions with Play Store don't think can be rooted.



> > seems to work but the Java part seems not unaliased:
> 
> Some time ago, I had read about it on the chromium-dev mailing list, but as it didn't happen to me I didn't investigate. try looking there.

So this works for you but only because you are using debug builds?

> no, unfortunately I'm still on your second version. do you have any tools that compare applied patches and are able to produce some sort of diff to apply?
> at most I throw everything and start again (for the third time :), I'll do some other pulls (sorry if I clog the repo) so I download the changes/new patches I'm doing.

I have made separate commits, you can see the diffs there - commit by commit. I am surprised that previous versions would not crash too.

> maybe you have removed too much.

It must be done, otherwise it is just Chromium with some whistles and bells.

uazo commented:
> It says adbd cannot run as root in production builds; the versions with Play Store don't think can be rooted.

just tried it, it works, they are userdebug

![image](https://user-images.githubusercontent.com/29201891/178135947-9badb4d1-34da-4cf7-95f8-dd3d71cf09e6.png)

> So this works for you but only because you are using debug builds?

I'm sorry, I don't know.

> I have made separate commits, you can see the diffs there - commit by commit

sure, but it is rather cumbersome and not free from errors. it would be easier if we shared only the modified files during the rebase, so as to reapply them more easily.
@chirayudesai these things would be useful, as well as others

> I am surprised that previous versions would not crash too.

that's how it seems

> I launch the build, but it takes at least 3 hours.

sorry, i could only now, this is the crash that I detect:
```
uild fingerprint: 'Android/sdk_phone_x86_64/generic_x86_64:9/PSR1.180720.012/4923214:userdebug/test-keys'
Revision: '0'
pid: 20097, tid: 20097, name: ite.bromite.dev  >>> org.bromite.bromite.dev <<<
signal 5 (SIGTRAP), code 128 (SI_KERNEL), fault addr 0x0
[FATAL:jni_android.cc(306)] Please include Java exception stack in crash report'
Abort message: '[FATAL:jni_android.cc(306)] Please include Java exception stack in crash report'

Stack Trace:
  RELADDR   FUNCTION                                                                          FILE:LINE
  0000000008efcdb5  logging::LogMessage::~LogMessage()                                                ../../base/logging.cc:925:7
  0000000008fcd3b3  base::android::CheckException(_JNIEnv*)                                           ../../base/android/jni_android.cc:306:3
  v------>  jni_generator::CheckException(_JNIEnv*)                                           ../../base/android/jni_generator/jni_generator_helper.h:53:3
  0000000004d4fb0e  jni_generator::JniJavaCallContextChecked::~JniJavaCallContextChecked()            ../../base/android/jni_generator/jni_generator_helper.h:118:5
  v------>  signin::Java_ProfileOAuth2TokenServiceDelegate_Constructor(_JNIEnv*, long, base::android::JavaRef<_jobject*> const&)  gen/components/signin/public/android/jni_headers/ProfileOAuth2TokenServiceDelegate_jni.h:108:1
  0000000005ad8076  ProfileOAuth2TokenServiceDelegateAndroid::ProfileOAuth2TokenServiceDelegateAndroid(AccountTrackerService*)  ../../components/signin/internal/identity_manager/profile_oauth2_token_service_delegate_android.cc:159:7
  0000000005ad4f50  std::__1::__unique_if<ProfileOAuth2TokenServiceDelegateAndroid>::__unique_single std::__1::make_unique<ProfileOAuth2TokenServiceDelegateAndroid, AccountTrackerService*&>(AccountTrackerService*&)  ../../buildtools/third_party/libc++/trunk/include/__memory/unique_ptr.h:725:32
  v------>  (anonymous namespace)::CreateAndroidOAuthDelegate(AccountTrackerService*)         ../../components/signin/internal/identity_manager/profile_oauth2_token_service_builder.cc:44:10
  v------>  (anonymous namespace)::CreateOAuth2TokenServiceDelegate(AccountTrackerService*, signin::AccountConsistencyMethod, SigninClient*, network::NetworkConnectionTracker*)  ../../components/signin/internal/identity_manager/profile_oauth2_token_service_builder.cc:121:10
  0000000005ad4eae  BuildProfileOAuth2TokenService(PrefService*, AccountTrackerService*, network::NetworkConnectionTracker*, signin::AccountConsistencyMethod, SigninClient*)  ../../components/signin/internal/identity_manager/profile_oauth2_token_service_builder.cc:180:7
  0000000005ab5da9  signin::BuildIdentityManagerInitParameters(signin::IdentityManagerBuildParams*)   ../../components/signin/public/identity_manager/identity_manager_builder.cc:112:7
  0000000005ab6170  signin::BuildIdentityManager(signin::IdentityManagerBuildParams*)                 ../../components/signin/public/identity_manager/identity_manager_builder.cc:198:7
  0000000008d49aa6  IdentityManagerFactory::BuildServiceInstanceFor(content::BrowserContext*) const   ../../chrome/browser/signin/identity_manager_factory.cc:160:7
  000000000a33d3fa  KeyedServiceFactory::GetServiceForContext(void*, bool)                            ../../components/keyed_service/core/keyed_service_factory.cc:80:15
  0000000008d44381  AccountInvestigatorFactory::BuildServiceInstanceFor(content::BrowserContext*) const  ../../chrome/browser/signin/account_investigator_factory.cc:41:28
  000000000a33d3fa  KeyedServiceFactory::GetServiceForContext(void*, bool)                            ../../components/keyed_service/core/keyed_service_factory.cc:80:15
  000000000a33c4c0  DependencyManager::CreateContextServices(void*, bool)                             ../../components/keyed_service/core/dependency_manager.cc:0:0
  000000000b19a1ed  BrowserContextDependencyManager::DoCreateBrowserContextServices(content::BrowserContext*, bool)  ../../components/keyed_service/content/browser_context_dependency_manager.cc:46:22
  0000000008d0ed58  ProfileImpl::OnLocaleReady(Profile::CreateMode)                                   ../../chrome/browser/profiles/profile_impl.cc:1101:51
  0000000008d0d6ba  ProfileImpl::OnPrefsLoaded(Profile::CreateMode, bool)                             ../../chrome/browser/profiles/profile_impl.cc:1142:3
  0000000008d0d477  ProfileImpl::ProfileImpl(base::FilePath const&, Profile::Delegate*, Profile::CreateMode, base::Time, scoped_refptr<base::SequencedTaskRunner>)  ../../chrome/browser/profiles/profile_impl.cc:524:5
  0000000008d0cae7  Profile::CreateProfile(base::FilePath const&, Profile::Delegate*, Profile::CreateMode)  ../../chrome/browser/profiles/profile_impl.cc:360:59
  0000000008d131b7  ProfileManager::CreateProfileHelper(base::FilePath const&)                        ../../chrome/browser/profiles/profile_manager.cc:1445:10
  0000000008d116c3  ProfileManager::CreateAndInitializeProfile(base::FilePath const&)                 ../../chrome/browser/profiles/profile_manager.cc:1926:38
  0000000008d10c4e  ProfileManager::GetProfile(base::FilePath const&)                                 ../../chrome/browser/profiles/profile_manager.cc:787:10
  0000000008d1135d  ProfileManager::CreateInitialProfile()                                            ../../chrome/browser/profiles/profile_manager.cc:762:24
```
now I look at it

uazo commented:
you need this https://gist.github.com/uazo/aaa68a31ca1d40a1a34e32b36865963c so it starts. 
unfortunately now crashes at the first navigation ...

csagan5 commented:
> you need this https://gist.github.com/uazo/aaa68a31ca1d40a1a34e32b36865963c so it starts.

Without this change in current & past versions of Bromite it was doing some attempt to retrieve an OAuth2 token, but also failing on the C++ side, I assume?



> unfortunately now crashes at the first navigation ...

So it opens but then after browsing crashes? I restarted the build in debug mode yesterday so I will soon be able to continue.



> just tried it, it works, they are userdebug

Yes, those are without Play Store. I can try without that but I have to figure out how to use the debug symbol with non-debug builds, because we release them and they should be useful for something.

uazo commented:
> So it opens but then after browsing crashes?

yes, the problem is "Allow building without enable_reporting"
so you fix it:
```
diff --git a/content/browser/renderer_host/render_frame_host_impl.cc b/content/browser/renderer_host/render_frame_host_impl.cc
index bdda609d5c..f229a90246 100644
--- a/content/browser/renderer_host/render_frame_host_impl.cc
+++ b/content/browser/renderer_host/render_frame_host_impl.cc
@@ -11437,7 +11437,6 @@ void RenderFrameHostImpl::TakeNewDocumentPropertiesFromNavigation(
   early_hints_manager_ = navigation_request->TakeEarlyHintsManager();

   // Only take some properties if this is not the synchronous initial
-#if BUILDFLAG(ENABLE_REPORTING)
   // `about:blank` navigation, because the values set at construction time
   // should remain unmodified.
   if (!navigation_request->IsWaitingToCommit()) {
@@ -11550,25 +11549,6 @@ void RenderFrameHostImpl::MaybeGenerateCrashReport(
   auto body = base::DictionaryValue();
   if (!reason.empty())
     body.SetString("reason", reason);
-
-  // Send the crash report to the Reporting API.
-  GetProcess()->GetStoragePartition()->GetNetworkContext()->QueueReport(
-      /*type=*/"crash", /*group=*/"default", last_committed_url_,
-      GetReportingSource(), isolation_info_.network_isolation_key(),
-      absl::nullopt /* user_agent */, std::move(body));
-#else
-}
-
-void RenderFrameHostImpl::OnSameDocumentCommitProcessed(
-    const base::UnguessableToken& navigation_token,
-    bool should_replace_current_entry,
-    blink::mojom::CommitResult result) {
-}
-
-void RenderFrameHostImpl::MaybeGenerateCrashReport(
-    base::TerminationStatus status,
-    int exit_code) {
-#endif
 }

 void RenderFrameHostImpl::SendCommitNavigation(
```

csagan5 commented:
> missing however [this](https://gist.github.com/uazo/c8e707d31fc96401cc238b62ceeb9182)

So, this was due to the preferences import; I have added it and will switch all builds to `treat_warnings_as_errors=true`.

> you need this https://gist.github.com/uazo/aaa68a31ca1d40a1a34e32b36865963c so it starts.

I have added this, except for the Java part because it is not compiled.

> yes, the problem is "Allow building without enable_reporting"
> so you fix it:

I have also added this, thanks. I was finally able to see the stacktrace as well.

uazo commented:
> anything else missing from your patches that would be beneficial for v103?

today I look at it, I'll let you know

> Do you see any problem due to the lack of this patch?

I don't know, I look at this too, along with https://github.com/bromite/bromite/pull/2157#discussion_r918408746
Is there anything else I haven't answered?

uazo commented:
in the meantime, I am making the list of v103 flags available to you, I think there is something to disable.
https://github.com/uazo/bromite-flags-list/commit/158aa14d1bd8c24f536d9a94855bd6626e63df3b

csagan5 commented:
> in the meantime, I am making the list of v103 flags available to you, I think there is something to disable.
> [uazo/bromite-flags-list@158aa14](https://github.com/uazo/bromite-flags-list/commit/158aa14d1bd8c24f536d9a94855bd6626e63df3b)

We can add guards for those already disabled, as per this one:
> # Enables configuring the on-device clustering backend.
> kOnDeviceClustering=0 # actually enabled

I think it still does nothing? e.g. no communication with third-party servers

uazo commented:
>> #Enables configuring the on-device clustering backend.
>> kOnDeviceClustering=0 # actually enabled

> I think it still does nothing? e.g. no communication with third-party servers

no communication, or rather, for those who have chrome with sync the data is sent to google, but in android everything is [disabled](https://source.chromium.org/chromium/chromium/src/+/main:components/history_clusters/core/features.cc;l=19;bpv=1;bpt=0)
